src/ng/q.js

```diff
@@ -254,25 +256,30 @@ function qFactory(nextTick, exceptionHandler) {
    *
    * @returns {Deferred} Returns a new instance of deferred.
    */
-  var defer = function() {
-    var d = new Deferred();
+  var defer = function(trackPromise) {
+    var d = new Deferred(trackPromise);
     //Necessary to support unbound execution :/
     d.resolve = simpleBind(d, d.resolve);
     d.reject = simpleBind(d, d.reject);
     d.notify = simpleBind(d, d.notify);
     return d;
-  };
+  }
 
-  function Promise() {
+  function Promise(trackPromise) {
     this.$$state = { status: 0 };
+  
+    this.trackPromise = typeof(trackPromise) !== 'undefined' ? trackPromise : true;
+    if (this.trackPromise) {
+      window.promises.angular.pendingCount++;
+    }
```

Comment:
Rather than doing it like this, 
I would suggest you encapsulate away the implementation of what trackPromise should do into a callback. 

Pass two callbacks, trackPromise, untrackPromise to the Deferred() constructor .... save those in 
this.trackPromise, and this.untrackPromise.
Inside that increment pendingCount. 

This would effectively shield the implementation of $q from the implementation of how we track the promises ..... Will also make the code easier to read.

Reply:
Also see if you can pass in the trackPromise, untrackPromise methods as part of the implementation of $qProvider, similar to how the exceptionHandler is setup 
This way, we can just setup the trackers for the promises once and be done with it.

Reply:
Done. I added a `$trackPromiseHandler` and `$untrackPromiseHandler` services that can be overridden by the user for tracking purposes instead. By default, those are no-op meaning they do not track any promises.